### PR TITLE
expression: implement vectorized evaluation for `builtinIntIsTrueSig`

### PR DESCRIPTION
**Pull request by liuhao2050**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinIntIsTrueSig. #12103

### What is changed and how it works?
```
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinOpFunc/builtinIntIsTrueSig-VecBuiltinFunc-4             339927              3433 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinOpFunc/builtinIntIsTrueSig-NonVecBuiltinFunc-4           47221             25373 ns/op               0 B/op          0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_op_vec.go [hunk 1]
```diff
@@ -382,11 +382,31 @@ func (b *builtinDecimalIsTrueSig) vecEvalInt(input *chunk.Chunk, result *chunk.C
...
+	defer b.bufAllocator.put(buf)
+
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(numRows, false)
+	i64s := result.Int64s()
+	for i := 0; i < numRows; i++ {
+		if buf.IsNull(i) || buf.GetInt64(i) == 0 {
```
Comment (Reminiscent) [inline expression/builtin_op_vec.go]: What about using the array other than ```buf.GetInt64(i) == 0```?

**Comment by qw4990** [comment]
Please resolve these conflicts.

File: expression/builtin_op_vec.go [hunk 2]
```diff
@@ -404,11 +404,32 @@ func (b *builtinDecimalIsTrueSig) vecEvalInt(input *chunk.Chunk, result *chunk.C
...
 
 func (b *builtinIntIsTrueSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinIntIsTrueSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	numRows := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, numRows)
```
Comment (XuHuaiyu) [inline expression/builtin_op_vec.go]: Can we reuse `result` instead of using a new `buf`?

Comment (liuhao2050) [reply expression/builtin_op_vec.go]: yes, we can do it in `result` and I have removed to use a new buffer.

File: expression/builtin_op_vec.go [hunk 3]
```diff
@@ -518,11 +518,26 @@ func (b *builtinDecimalIsTrueSig) vecEvalInt(input *chunk.Chunk, result *chunk.C
...
+	if err := b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+	i64s := result.Int64s()
+	for i := 0; i < numRows; i++ {
+		if result.IsNull(i) {
+			i64s[i] = 0
+			result.SetNull(i, false)
+		} else if i64s[i] == 0 {
+			i64s[i] = 0
```
Comment (SunRunAway) [inline expression/builtin_op_vec.go]: Is it a redundant operation here?

Comment (liuhao2050) [reply expression/builtin_op_vec.go]: yes, I will remove it.

**Pull request merged**